src/darwin/Framework/CHIP/MTRDeviceControllerFactory.h

```diff
@@ -149,6 +149,22 @@ API_AVAILABLE(ios(16.4), macos(13.3), watchos(9.4), tvos(16.4))
 - (MTRDeviceController * _Nullable)createControllerOnExistingFabric:(MTRDeviceControllerStartupParams *)startupParams
                                                               error:(NSError * __autoreleasing *)error;
 
+/**
+ * Create an additional MTRDeviceController on an existing fabric.  Returns nil on failure.
+ *
+ * The fabric is identified by the root public key and fabric id in
+ * the startupParams.
+ *
+ * This method will fail if there is no such fabric.
+ *
+ * This method will fail if the MTRDeviceControllerStartupParams don't follow
+ * the rules for creating a controller on a new fabric.  In particular, the
+ * vendor ID must not be nil.
+ */
+- (MTRDeviceController * _Nullable)createAdditionalControllerOnExistingFabric:(MTRDeviceControllerStartupParams *)startupParams
```

Comment:
Rather than creating another new method here, I'd create a second method with the same signature, except add something before error like: "secondary:(BOOL)secondary" that does the right thing. Cascading methods like this are more normal in ObjectiveC

Reply:
Right, this comes down to what the semantics we want are.  If it's a "create controller on existing fabric, optionally secondary", what you suggest makes sense and I will switch to it if we keep these semantics.

If we change to "just create a controller, without doing fabric checks" semantics-wise, then it's not really parallel to either of the two things we have (though if we had thought about it up front we could have had `createControllerWithParams:fabricCheck:error:` or something where `fabricCheck:` takes an enum describing the behavior....

Reply:
Creating a new init method with a vague name isn't great. We should keep all existing semantics with the current one, and you can add an enum for fabricCheck (better name? fabricBehavior:?) that by default the current one does what it does already.